### PR TITLE
Replace deprecated ageable with breedable

### LINKED ISSUE #4164: Update `#setAgeLock()` and `#getAgeLock` deprecation in `ReplicatingEntityWrapper`
https://github.com/IntellectualSites/PlotSquared/blob/b7c9453a1ad3bad8cce37487607854e4db535f17/Bukkit/src/main/java/com/plotsquared/bukkit/entity/ReplicatingEntityWrapper.java#L384

instead of calling `Ageable` class, `Breedable` should be used in this method. Make sure to check update casts, if necessary.

https://github.com/IntellectualSites/PlotSquared/blob/b7c9453a1ad3bad8cce37487607854e4db535f17/Bukkit/src/main/java/com/plotsquared/bukkit/entity/ReplicatingEntityWrapper.java#L394

this method requires attention. Given it's public, we can't just change it without causing a breaking API change, which is undesired.
Rather, the method should be deprecated for removal and a new method calling the proper class has to be introduced.
The new introduced method has to replace the deprecated calls in the same class.

### PR DESCRIPTION
**Pull request by TheMeinerLP**
## Overview
Replace depreacted ageable.

Fixes #4164 

## Description

```[tasklist]
### Submitter Checklist
- [X] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right sidX)) and not your main branch.
- [X] Ensure that the pull request title represents the desired changelog entry.
- [X] New public fields and methods are annotated with `@since TODO`.
- [X] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).
```

### CODE REVIEW

**Comment by dordsor21** [comment]
Missing since tags on the deprecated and the new methods

**Review by NotMyFault — COMMENTED** [review]
since's should be TODO, and deprecations have to have since's too.

**Comment by TheMeinerLP** [comment]
> since's should be TODO, and deprecations have to have since's too.

Is there a useful adapter ?

**Comment by NotMyFault** [comment]
> > since's should be TODO, and deprecations have to have since's too.
> 
> Is there a useful example ?

https://www.baeldung.com/java-deprecated#optional-attributes

File: Bukkit/src/main/java/com/plotsquared/bukkit/entity/ReplicatingEntityWrapper.java [hunk 1]
```diff
@@ -392,13 +398,41 @@ private void restoreAgeable(Ageable entity) {
...
+        entity.setAgeLock(this.aged.locked);
+        if (this.aged.age > 0) {
+            entity.setAge(this.aged.age);
+        }
+    }
+
+    /**
+     * @since TODO
+     */
+    public void storeBreedable(Breedable breedable) {
```
Comment (SirYwell) [inline Bukkit/src/main/java/com/plotsquared/bukkit/entity/ReplicatingEntityWrapper.java:429]: Can be private

**Review by SirYwell — CHANGES_REQUESTED** [review]

**Review by SirYwell — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
Thanks!

**Pull request merged**